tests/unit/formatDefects.unitTest.ts

```diff
@@ -0,0 +1,196 @@
+import { convertFlatDataToProperJSON } from "../../src/utils/formatDefects";
+
+describe("convertFlatDataToProperJSON", () => {
+  it("should convert flat data to proper JSON", () => {
+    const flatData = [
+      {
+        sectionNumber: "01",
+        sectionDescription: "Noise",
+        vehicleTypes_0: "Cars",
+        euVehicleCategories_0: "M1",
+        requiredStandards_0_rsNumber: 1,
+        requiredStandards_0_requiredStandard:
+          "The exhaust must be securely mounted.",
+        requiredStandards_0_refCalculation: "1.1",
+        requiredStandards_0_additionalInfo: true,
+        requiredStandards_0_inspectionTypes_0: "basic",
+        requiredStandards_1_rsNumber: 2,
+        requiredStandards_1_requiredStandard:
+          "The exhaust must have all components secure.",
+        requiredStandards_1_refCalculation: "1.2",
+        requiredStandards_1_additionalInfo: true,
+        requiredStandards_1_inspectionTypes_0: "basic",
+      },
+    ];
+
+    const expectedResult = [
```

Comment:
Is it worth checking the output against the type defined in the types package rather than raw JSON?